### DISCUSSION TITLE
IGazeControl and PositionControl in the same application

### DISCUSSION THREAD

**Discussion by ddetommaso** [robotology/yarp#3072] [Support - Q&A] [answered]
I would like to control the head of the robot and I am able to do it using the IGazeControl and the IPositionControl separately. 
But if I want to use both controllers within the same application this seems not possible cause after using the IGazeControl, it takes control over on the joints of the head and any later attempt of moving the same joints with the IPositionControl will fail.

![image](https://github.com/robotology/yarp/assets/3440640/66df4ed8-957b-4add-97b4-5cf2434eadd1)

Is there any command to make the IGazeControl release its control on the joints ?

**Comment by pattacini** [comment, marked as answer]
Hi @ddetommaso 

>Is there any command to make the IGazeControl release its control on the joints ?

It's not a matter of releasing the control. Rather, you ought to switch the control mode of the part involved explicitly before using it.

To this end, refer to [`setControlMode`](https://yarp.it/latest/classyarp_1_1dev_1_1IControlMode.html#a0ea675c5713f24d1cdb319b2eadb794e).

**Reply by ddetommaso** [reply]
Hi @pattacini 

that worked, thank you so much!!
I was setting the control mode during the initialization of the position controllers, not always before using them. I think my confusion stems from the fact that there is no explicit control mode setting for the gaze controller, so for this controller is not actually needed to use the setControlMode.

**Reply by pattacini** [reply]
>o for this controller is not actually needed to use the setControlMode.

Yep, the gaze controller does this internally.
